libr/util/hex.c

```diff
@@ -450,6 +450,31 @@ R_API int r_hex_str2bin(const char *in, ut8 *out) {
 	return nibbles / 2;
 }
 
+// get the hex chars from start of string, until first non-hex char, as a heap
+// allocated ut8* buffer
+R_API int r_hex_str2bin_until_new(const char *in, ut8 **out) {
+	r_return_val_if_fail (in && out && !*out, -1);
+	size_t len = strlen (in);
+	if (len <= 1) {
```

Comment:
this only happens if the string is empty. so no need to strlen in this case, do u think that *in should be checked in the return_if_fail statement?

also, there's R_STR_ISEMPTY()) and R_STR_ISNOTEMPTY() macros to help ont hat

Reply:
I think the way it is makes life a bit easier on the caller. I might misunderstand you though.

I think `in == NULL` is indicative of a programmer error and so `r_return_val_if_fail`. The case `*out != NULL` is bad because the caller should be able to free `*out` regardless of success/failure/error. This means the caller does not have to handle as many edge cases. I could see a case for `out == NULL` being accepted, just to let the function count how many bytes of hex it would parse.

I think if `in` is "", "4", "ZZZZ" or "4Z" then the function should return 0 because it did not parse any hex from the start of the string. This shouldn't be a programing error. This input will happen when trying to get a byte mask from the invalid signatures "11:", "11:4", "11:ZZZZ" and "11:4Z". So the caller has less edge cases to handle.

So input strings of length 1 or 0 should return 0, since neither can have any hex *bytes* parsed. This function can't parse nibbles otherwise the caller wouldn't be able to know where the parser stopped in the string (via: `in + (2 * return_code)`)